lib/cast.js

```diff
@@ -0,0 +1,132 @@
+const ColumnCompiler_PG = require('./dialects/postgres/schema/pg-columncompiler');
+const Raw = require('./raw');
+const ColumnCompiler_MySQL = require('./dialects/mysql/schema/mysql-columncompiler');
+const ColumnCompiler_MSSQL = require('./dialects/mssql/schema/mssql-columncompiler');
+const ColumnCompiler_Oracledb = require('./dialects/oracledb/schema/oracledb-columncompiler');
+const ColumnCompiler_Redshift = require('./dialects/redshift/schema/redshift-columncompiler');
+const ColumnCompiler_SQLite3 = require('./dialects/sqlite3/schema/sqlite-columncompiler');
+const {
+  augmentWithBuilderInterface,
+} = require('./builder-interface-augmenter');
+const helpers = require('./util/helpers');
+
+class Cast extends Raw {
+  constructor(client) {
+    super();
+    this.client = client;
+    this.value = null;
+    this.to = '';
+    this.alias = '';
+  }
+
+  cast(value, to, alias) {
+    this.value = value;
+    this.to = to;
+    this.alias = alias;
+    return this;
+  }
+
+  toString() {
+    return this.toQuery();
+  }
+
+  _getColType(type) {
+    let columnCompiler;
+
+    switch (this.client.config.client) {
```

Comment:
why not implement getCompiler method on this.client instead? not sure we don't have one yet, actually